core/deployment/src/main/java/io/quarkus/deployment/ide/IdeProcessor.java

```diff
@@ -0,0 +1,163 @@
+package io.quarkus.deployment.ide;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
+import io.quarkus.runtime.util.JavaVersionUtil;
+
+public class IdeProcessor {
+
+    private static Map<String, List<Ide>> IDE_MARKER_FILES = new HashMap<>();
+    private static Map<Predicate<ProcessInfo>, Ide> IDE_PROCESSES = new HashMap<>();
+
+    static {
+        IDE_MARKER_FILES.put(".idea", Collections.singletonList(Ide.IDEA));
+        IDE_MARKER_FILES.put(".project", Arrays.asList(Ide.VSCODE, Ide.ECLIPSE));
+        IDE_MARKER_FILES = Collections.unmodifiableMap(IDE_MARKER_FILES);
+
+        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("idea")), Ide.IDEA);
+        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("code")), Ide.VSCODE);
+        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("eclipse")), Ide.ECLIPSE); // TODO: I have no idea if this is correct
+        IDE_PROCESSES = Collections.unmodifiableMap(IDE_PROCESSES);
+    }
+
+    @BuildStep
+    public EffectiveIdeBuildItem effectiveIde(IdeConfig ideConfig, IdeFileBuildItem ideFile,
+            IdeRunningProcessBuildItem ideRunningProcess) {
+        Ide result = null;
+        if (ideConfig.target == IdeConfig.Target.auto) {
+
+            // the idea here is to auto-detect the special files that IDEs create
+            // and also the running IDE process if need be
+
+            if (ideFile.getDetectedIDEs().size() == 1) {
+                result = ideFile.getDetectedIDEs().iterator().next();
+            } else {
+                Set<Ide> runningIdes = ideRunningProcess.getDetectedIDEs();
+                if (runningIdes.size() == 1) {
+                    result = runningIdes.iterator().next();
+                } else {
+                    List<Ide> matches = new ArrayList<>();
+                    for (Ide file : ideFile.getDetectedIDEs()) {
+                        for (Ide process : runningIdes) {
```

Comment:
oh wow  - just realized you detecting running ides ....thats awesome :)

Reply:
one caveat - this requires the IDE to be running when we start quarkus:dev, right?

also - while testing this having no IDE running i stopped getting links highlighted since no changes made and thus we never relooked for running processes ?

feels a bit weird.

Reply:
and changing code to try trigger recalculation of running ides does not seem to be enough.

Reply:
People close their IDE?

Reply:
I do constantly :)

Reply:
The IDE detection is necessary because there from the project files we can't really diferentiate if VSCode is Eclipse is being used.

Reply:
If there is another way to differentiate an Eclise project from a VSCode project, then I'll gladly use it.

Reply:
> rigger recalculation of runni

When the changes result in a Quarkus dev restart, the processes are re-detected.

Reply:
Weird - it stopped showing links for me :/